library/config.c

```diff
@@ -60,8 +60,12 @@ int ziti_load_config(ziti_config *cfg, const char* cfgstr) {
     }
 
     if (rc < 0) {
-        free_ziti_config(cfg);
+        return rc;
```

Comment:
should free_ziti_config still be called?

Reply:
and actually, shouldn't this whole block just move up to 57-60? it's the same if statement

Reply:
oooh i see. it's testing it twice and setting it ... nevermind ☝️

Reply:
no, `rc` changes between 57 and here